actors/power/src/state.rs

```diff
@@ -104,25 +104,29 @@ impl State {
         policy: &Policy,
         s: &BS,
         miner: &Address,
-    ) -> anyhow::Result<bool> {
+    ) -> anyhow::Result<(StoragePower, bool)> {
         let claims = make_map_with_root_and_bitwidth(&self.claims, s, HAMT_BIT_WIDTH)?;
 
-        let claim =
-            get_claim(&claims, miner)?.ok_or_else(|| anyhow!("no claim for actor: {}", miner))?;
+        let claim = match get_claim(&claims, miner)? {
+            Some(c) => c,
+            None => {
+                return Ok((StoragePower::zero(), false));
```

Comment:
This was better failing if the address is not a miner. There is a difference between a miner with zero power and a non-miner actor. In fact, the miner_raw_power method is probably how external parties will detect that.

Reply:
This raises the spectre of `Anyhow` – I resisted the urge to ask you to purge it here so as to return an `ActorResult`, but in this case we probably want to return IllegalArgument instead of IllegalState. So would you replace the `?` with `[with_]context_code()` calls?

Reply:
Yup, I think I agree, thank you.